### ISSUE TITLE
Misleading data curation and sorting

### ISSUE THREAD

**Issue by sermare** [openvax/mhcflurry#250]
I have collected the IEDB Database, then downloaded the files from the paper (S2/S3) and used the GitHub to download both the training/benchmark data. At first glance, the testing data is correctly independent of the training data. 

![Image](https://github.com/user-attachments/assets/b2f6e91f-14c8-490e-94d6-a45104b8d869)

However, when evaluating the model on their predictions I get suspiciously a high correlation. 

![Image](https://github.com/user-attachments/assets/568eb3f2-15b9-46ee-b75b-844a421da6f7)

When I evaluate the model against peptides submitted after 2021 (MHC Flurry published in 2020), I get a more realistic spearman:

![Image](https://github.com/user-attachments/assets/986f5dea-1388-40d2-baf3-5f48e6f8b07d)

Could you help me understand what has been done with the model ever since publication? Was the model fine-tuned on the rest of the data and published? Is the previous evaluation data, part of your training data? How do you expect to benchmark this model against other models?

**Comment by timodonnell** [comment]
I'm not sure I totally follow here - but if I'm reading the axes right in the first plot you are showing the prediction of one mhcflurry variant vs another (i.e. the "no_additional_ms" predictor that had a lot of monoallelic MS data held out so we could test on it, vs. the published predictor that includes this data in its train set), whereas in the second plot you are showing mhcflurry prediction vs. actual experimental IC50s. Is that right?

To me it doesn't seem too surprising that the two mhcflurry predictor variants would be more correlated with each other than either of them against ground truth.

In terms of what's been done with the models post publication - for better for worse, nothing has changed for them. If you load a model you can actually see when it was trained by looking at str(model), e.g.:

<img width="889" alt="Image" src="https://github.com/user-attachments/assets/59931df3-6afc-4d2c-912b-7d21b6f56647" />

**Comment by sermare** [comment]
I've clarified my misunderstanding and I forgot to mention this data is subset to only HLA-A*02:01. The test set (Supplement 2) only includes monoallelic MS data with binary ground truth labels (0 for non-binders, 1 for binders), and it does not contain any actual binding affinity measurements. In the first plot, I’m simply comparing predictions from one model against those from another, not against binding affinity values as I initially assumed. Meanwhile, the bottom plot shows predictions on post‑2021 IEDB epitopes, yielding a Spearman correlation of approximately 0.4, which aligns well with the correlations reported in other studies. My las question is, what is the difference between the training/test set from the paper (which is accessed by https://data.mendeley.com/datasets/zx3kjzc3yx/3) and the training/test data accessed by the GitHub? In the first plot I show that there is overlap but a large number of peptides are unique to a set. I have the data "REPO" referring to the data accessed through the github.
